packages/frontend/src/services/tokenExchange/FungibleTokenExchange.js

```diff
@@ -278,25 +278,39 @@ class FungibleTokenExchange {
 
         return txs.length ? txs : null;
     }
-    // @todo handle each transaction and break
-    // subsequent TXs in case of previous one is failed;
-    // return informative message in such situations
+
     async _processTransactions(account, txs) {
         const swapResult = {
             success: true,
             swapTxHash: '',
+            failReason: '',
         };
 
         for (const tx of txs) {
             const {
+                status,
+                receipts_outcome,
                 transaction: { hash, actions },
             } = await account.signAndSendTransaction(tx);
 
-            const swapTx = actions.find((action) => {
-                if (action['FunctionCall']?.method_name === 'ft_transfer_call') {
-                    return true;
-                }
-            });
+            const failedResult = receipts_outcome.find(
+                ({ outcome: { status } }) => !!status?.Failure
+            );
+
+            if (status?.Failure) {
+                swapResult.success = false;
```

Comment:
`swapResult.success = !!status?.Failure;`

Might be hard to read but shorter

Reply:
`!status?.Failure`

Reply:
`!!` if `status?.Failure` is a string. If not then `!` suppose will be enough

Reply:
I don't get it. We have a success result if it's not `status?.Failure`. It doesn't matter which type there is (but in general it has to be an object). So `!status?.Failure` is enough, no?